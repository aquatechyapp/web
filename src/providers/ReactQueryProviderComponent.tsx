'use client';

import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { ReactQueryDevtools } from '@tanstack/react-query-devtools';
import { useState } from 'react';

import pj from '../../package.json';

export function ReactQueryProviderComponent({ children }: { children: React.ReactNode }) {
  console.log({
    env: process.env.NODE_ENV,
    version: pj.version
  });

  const [queryClient] = useState(
    () =>
      new QueryClient({
        defaultOptions: {
          queries: {
            staleTime: 60 * 1000,
            refetchOnWindowFocus: false,
            retry: false,
            gcTime: 10 * 60 * 1000 // 10 minutes
          }
        }
      })
  );

  return (
    <QueryClientProvider client={queryClient}>
      {children}
      <ReactQueryDevtools />
    </QueryClientProvider>
  );
}
