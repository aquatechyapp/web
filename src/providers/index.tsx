'use client';

import { LoadingProvider } from '@/context';

export const Providers = ({ children }: { children: React.ReactNode }) => {
  return <LoadingProvider>{children}</LoadingProvider>;
};
