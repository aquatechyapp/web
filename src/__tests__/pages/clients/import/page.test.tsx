import { render } from '@testing-library/react';

import { ReactQueryMocked } from '@/__tests__/providers/reactQueryMocked';
import ClientsImportPages from '@/app/(authenticated)/clients/import/page';

vi.mock('useMutation', () => ({
  mutate: vi.fn()
}));

describe('/clients/import', () => {
  const wrapper = render(
    <ReactQueryMocked>
      <ClientsImportPages />
    </ReactQueryMocked>
  );

  it('should render the ImportClientsPage component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
