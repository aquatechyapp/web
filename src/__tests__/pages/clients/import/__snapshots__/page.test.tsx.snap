// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`/clients/import > should render the ImportClientsPage component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <div
          class="rounded-md border"
        >
          <div
            class="mx-2 my-4 flex w-fit flex-wrap gap-4 text-nowrap md:flex-nowrap"
          >
            <button
              class="inline-flex items-center justify-center gray-50space-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-slate-950 disabled:pointer-events-none disabled:opacity-50 dark:focus-visible:ring-slate-300 bg-slate-900 text-slate-50 shadow hover:bg-slate-900/90 dark:bg-slate-50 dark:text-slate-900 dark:hover:bg-slate-50/90 h-9 px-4 py-2"
              type="button"
            >
              <a
                download="import-sample-aquatechy"
                href="/sample-import-aquatechy.csv"
              >
                Download Sample File
              </a>
            </button>
            <div
              class="space-y-2"
            >
              <input
                accept=".csv"
                aria-describedby=":r0:-form-item-description"
                aria-invalid="false"
                class="flex h-9 rounded-md border border-slate-200 bg-transparent px-3 py-1 text-sm shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-slate-500 focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-slate-950 disabled:cursor-not-allowed disabled:opacity-50 dark:border-slate-800 dark:placeholder:text-slate-400 dark:focus-visible:ring-slate-300 w-fit"
                id=":r0:-form-item"
                name="csvFile"
                placeholder="Picture"
                type="file"
              />
            </div>
          </div>
          <div
            class="flex flex-col gap-4"
          />
          <div
            class="w-full p-2"
          >
            <button
              class="inline-flex items-center justify-center gray-50space-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-slate-950 disabled:pointer-events-none disabled:opacity-50 dark:focus-visible:ring-slate-300 bg-slate-900 text-slate-50 shadow hover:bg-slate-900/90 dark:bg-slate-50 dark:text-slate-900 dark:hover:bg-slate-50/90 h-9 px-4 py-2 w-full"
              disabled=""
            >
              Import Clients
            </button>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <div
        class="rounded-md border"
      >
        <div
          class="mx-2 my-4 flex w-fit flex-wrap gap-4 text-nowrap md:flex-nowrap"
        >
          <button
            class="inline-flex items-center justify-center gray-50space-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-slate-950 disabled:pointer-events-none disabled:opacity-50 dark:focus-visible:ring-slate-300 bg-slate-900 text-slate-50 shadow hover:bg-slate-900/90 dark:bg-slate-50 dark:text-slate-900 dark:hover:bg-slate-50/90 h-9 px-4 py-2"
            type="button"
          >
            <a
              download="import-sample-aquatechy"
              href="/sample-import-aquatechy.csv"
            >
              Download Sample File
            </a>
          </button>
          <div
            class="space-y-2"
          >
            <input
              accept=".csv"
              aria-describedby=":r0:-form-item-description"
              aria-invalid="false"
              class="flex h-9 rounded-md border border-slate-200 bg-transparent px-3 py-1 text-sm shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-slate-500 focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-slate-950 disabled:cursor-not-allowed disabled:opacity-50 dark:border-slate-800 dark:placeholder:text-slate-400 dark:focus-visible:ring-slate-300 w-fit"
              id=":r0:-form-item"
              name="csvFile"
              placeholder="Picture"
              type="file"
            />
          </div>
        </div>
        <div
          class="flex flex-col gap-4"
        />
        <div
          class="w-full p-2"
        >
          <button
            class="inline-flex items-center justify-center gray-50space-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-slate-950 disabled:pointer-events-none disabled:opacity-50 dark:focus-visible:ring-slate-300 bg-slate-900 text-slate-50 shadow hover:bg-slate-900/90 dark:bg-slate-50 dark:text-slate-900 dark:hover:bg-slate-50/90 h-9 px-4 py-2 w-full"
            disabled=""
          >
            Import Clients
          </button>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
